lib/libzrepl/mgmt_conn.c

```diff
@@ -1280,8 +1300,10 @@ uzfs_zvol_mgmt_thread(void *arg)
 	mutex_destroy(&conn_list_mtx);
 
 	mutex_enter(&async_tasks_mtx);
-	(void) close(mgmt_eventfd);
-	mgmt_eventfd = -1;
+	if (mgmt_eventfd >= 0) {
+		(void) close(mgmt_eventfd);
+		mgmt_eventfd = -1;
+	}
 	while ((async_task = SLIST_FIRST(&async_tasks)) != NULL) {
 		SLIST_REMOVE_HEAD(&async_tasks, task_next);
 		uzfs_zinfo_drop_refcnt(async_task->zinfo, B_FALSE);
```

Comment:
See if we can use free_async_task() to free async_task as payload is not getting freed here as we are justing freeing
async_task.